crates/autopilot/src/arguments.rs

```diff
@@ -379,6 +379,13 @@ pub enum FeePolicyKind {
     Volume { factor: f64 },
 }
 
+fn validate_factor(factor: f64) -> Result<(), String> {
+    if !(0.0..1.0).contains(&factor) {
```

Comment:
This can be boiled down to a single function call.
```rust
anyhow::ensure!((0.0..1.0).contains("&factor"), "Factor must be in the range ...")
```
Alternatively we could consider introducing a new type `FeeFactor` here which enforces that in its constructor. Might be a bit overkill, though.